api/app/Models/PoolCandidate.php

```diff
@@ -535,8 +535,8 @@ public function createSnapshot()
             'essentialSkills',
             'nonessentialSkills',
         ])->findOrFail($this->pool_id);
-        $essentialSkillIds = $pool->essentialSkills()->pluck('id')->toArray();
-        $nonessentialSkillIds = $pool->nonessentialSkills()->pluck('id')->toArray();
+        $essentialSkillIds = $pool->essentialSkills()->pluck('skills.id')->toArray();
```

Comment:
Why this change? Doesn't essentialSkills return the same thing it did before?

Reply:
Upon adding id to the new pivot table in this commit, tests started failing https://github.com/GCTC-NTGC/gc-digital-talent/pull/7953/commits/cfe5c212d7b70d75c1f5fad86ffc0493a0ebc158

The failing tests seemed similar to me, they related to the snapshot 

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/e57c9c00-4740-42d8-a1eb-4955a144e1d3)

So I concluded that it was because calling

`pool->essentialSkills()->pluck('id')`

Was ambiguous on `id` due to `id` being present on `skills` and `pool_skill`
Specifying which table to pull id from resolved things.

Reply:
Huh, okay. I guess its because you're chaining methods on the relationship querybuilder directly, not on the resulting objects.